invenio_vocabularies/cli.py

```diff
@@ -8,76 +8,122 @@
 # details.
 
 """Commands to create and manage vocabulary."""
-
-import json
+import csv
+from os.path import dirname, join
 
 import click
 from flask.cli import with_appcontext
 from flask_principal import Identity
 from invenio_access import any_user
 from invenio_db import db
 
-from invenio_vocabularies.records.api import Vocabulary
-from invenio_vocabularies.records.models import VocabularyMetadata, \
-    VocabularyType
+from invenio_vocabularies.records.models import VocabularyType
 from invenio_vocabularies.services.service import VocabulariesService
 
+data_directory = join(dirname(__file__), "data")
+
+available_vocabularies = {
+    "languages": {
+        "path": join(data_directory, "languages.csv"),
+    },
+    "licenses": {
+        "path": join(data_directory, "licenses.csv"),
+    },
+}
+
+
+def _load_csv_data(path):
+    with open(path) as f:
+        reader = csv.DictReader(f, skipinitialspace=True)
+        dicts = [row for row in reader]
+        return dicts
+
+
+def _create_vocabulary(vocabulary_type_name, source_path):
+    identity = Identity(1)
+    identity.provides.add(any_user)
+    service = VocabulariesService()
+
+    # Load data
+    rows = _load_csv_data(source_path)
+
+    # Create vocabulary type
+    vocabulary_type = VocabularyType(name=vocabulary_type_name)
+    db.session.add(vocabulary_type)
+    db.session.commit()
+
+    i18n = ["title", "description"]  # Attributes with i18n support
+    other = ["icon"]  # Other top-level attributes
+
+    default_language = "en"  # Static (dependent on the files)
+
+    metadata = {"title": {}, "description": {}, "props": {}}
+
+    records = []
+    for row in rows:
+        for attribute in row:
+            value = row[attribute]
+            if attribute in i18n:
+                metadata[attribute][default_language] = value
+            elif any(map(lambda s: value.startswith(s + "_"), i18n)):
+                [prefix_attr, language] = attribute.split("_", 1)
+                metadata[prefix_attr][language] = value
+            elif attribute in other:
+                metadata[attribute] = value
+            else:
+                metadata["props"][attribute] = value
```

Comment:
Minor: I think the fact that the `id` key (e.g. for languages that would be `"en"`, `"fr"`, etc.) ends up in `props` instead of the top-level... But I guess there's a bigger discussion on how to manage the "user-friendly" identifier of the vocabularies inside the record metadata (or PID)

Reply:
It's intended that the `id` ends up in `props` (maybe should have named it `value` for distinction).